framework/web/User.php

```diff
@@ -562,6 +545,48 @@ protected function sendIdentityCookie($identity, $duration)
     }
 
     /**
+     * Determines if an identity cookie has a valid format and contains a valid auth key. 
+     * This method is used when [[enableAutoLogin]] is true.
+     * This method attempts to authenticate a user using the information in the identity cookie.
+     * @return array|null Returns an array of 'identity' and 'duration' if valid, otherwise null.
+     * @see loginByCookie()
+     */
+    protected function validateIdentityCookie()
+    {
+        $value = Yii::$app->getRequest()->getCookies()->getValue($this->identityCookie['name']);
+        if ($value === null) {
+            return;
+        }
+        $data = json_decode($value, true);
+        if (count($data) === 3 && isset($data[0], $data[1], $data[2])) {
```

Comment:
I changed my code to use Json::decode and did some testing.  If the contents of the cookie are malformed in some way and Json::decode is used, then an exception will be thrown.  When json_decode is used, my code recognizes that the contents are not usable, no cookie login is performed, and a standard login screen is presented.  I think the behavior using json_decode is preferable, so I change it back.